### PR TITLE
feat(idd): enforce mandatory F4 cleanup apply with explicit fallback paths

### LINKED ISSUE #430: feat(idd): enforce mandatory F4 cleanup apply before final release
## Overview
Require a deterministic F4 cleanup apply step in every merged PR flow so resolved feedback and stale operational markers are minimized immediately, not left as backlog.

## Problem
The merged-PR audit found many PRs in `needs-apply` status, meaning cleanup candidates were detected but never minimized. This drifts from the IDD requirement to hide resolved feedback and mission-complete markers.

## Scope
- Make F4 treat `audit-pr-cleanup --apply` as mandatory when candidates exist and permissions allow minimization.
- Require visible machine-readable and human-readable cleanup evidence in PR comments after merge.
- Define strict fallback behavior when minimization permissions are unavailable (explicit hold/escalation path, no silent skip).
- Add tests/docs covering required/skip/error paths.

## Acceptance Criteria
- [ ] F4 no longer exits cleanly when cleanup candidates exist but apply is skipped without an explicit blocker path.
- [ ] Merged PR flows include a deterministic cleanup evidence record (applied/skipped/failed summary + reason).
- [ ] Tests cover mandatory apply behavior and permission/error fallbacks.
- [ ] Documentation reflects the new mandatory cleanup contract.

## Non-goals
- Reclassifying cleanup candidates themselves.

## Refs
- Refs #72
- Refs #95
- Refs #324
- Refs #328

### PR DESCRIPTION
**Pull request by kurone-kito**
## Summary

Closes #430

F4 cleanup apply is now mandatory when candidates exist. The previous
"best-effort" approach silently skipped cleanup when it failed or when
the viewer lacked minimization permission, leaving merged PRs in
`needs-apply` backlog.

### Changes

**`scripts/audit-pr-cleanup-summary.mjs`**
- New `"permission-blocked"` dry-run status: emitted when no
  apply-eligible candidates exist but at least one skipped item has
  `viewerCanMinimize: false`. Allows F4 to distinguish "truly clean"
  from "items exist but viewer has no permission."

**`tests/audit-pr-cleanup-summary.test.mjs`**
- 6 new tests covering: `clean`, `permission-blocked`,
  `needs-apply` (with mixed viewer-cannot-minimize), mandatory `applied`
  path, `incomplete` (partial apply), `failed` (all-fail), and the
  mixed applied-and-failed case (pins that `failed` takes precedence).

**`.github/instructions/idd-merge.instructions.md` + template mirror**
- F4 decision tree keyed on dry-run `status`:
  - `clean` → no action
  - `needs-apply` → apply (mandatory); post one evidence comment with
    `status`, counts, `viewer-cannot-minimize` count when non-zero;
    use cleanup-failure comment format when apply is `failed`/`incomplete`
  - `permission-blocked` → post cleanup-permission-blocked comment;
    no silent exit
- GraphQL fallback path mirrors the same evidence-posting contract.

**`docs/idd-comment-minimization.md`**
- New "Mandatory F4 Cleanup Contract" section: decision-tree table,
  evidence comment format (with `<!-- idd-cleanup-evidence: ... -->`
  machine-readable token), cleanup-failure comment format, and
  cleanup-permission-blocked comment format.

### Non-goals

- Reclassifying cleanup candidates themselves (unchanged)
- Re-blocking a merge that already succeeded in F3

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

## Summary by CodeRabbit

* **Documentation**
  * Made F4 cleanup mandatory with a detailed decision tree and standardized evidence, failure, and permission-blocked comment templates; added guidance for dry-run, apply, and fallback flows.
* **Tests**
  * Added tests covering all dry-run/apply status outcomes and permission-edge cases.
* **Implementation**
  * Improved status detection to surface a permission-blocked state and enforce apply when candidates exist.

[Review Change Stack](https://app.coderabbit.ai/change-stack/kurone-kito/idd-skill/pull/447)

### CODE REVIEW

**Comment by kurone-kito** [comment]
_claude-sonnet-4-6: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
_claude-sonnet-4-6: critique baseline — IDD automation marker. Do not edit._

File: scripts/audit-pr-cleanup-summary.mjs [hunk 1]
```diff
@@ -15,7 +15,13 @@ export function computeReportSummary(report) {
   };
 
   if (report.mode === "dry-run") {
-    report.status = report.candidates.length === 0 ? "clean" : "needs-apply";
+    if (report.candidates.length > 0) {
+      report.status = "needs-apply";
+    } else if (viewerCannotMinimize > 0) {
+      report.status = "permission-blocked";
+    } else {
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply scripts/audit-pr-cleanup-summary.mjs:24]: **Accepted** — fixed in 261ba1b: changed `viewerCannotMinimize` filter to `!skip.isMinimized && !skip.viewerCanMinimize`, so already-minimized items no longer trigger `permission-blocked`. A regression test was also added.

File: tests/audit-pr-cleanup-summary.test.mjs [hunk 2]
```diff
@@ -56,3 +56,96 @@ test("computeReportSummary reflects apply results after mutations", () => {
...
+      { subjectId: "skip-2", isMinimized: false, viewerCanMinimize: false },
+    ],
+  });
+
+  computeReportSummary(report);
+
+  assert.equal(report.status, "permission-blocked");
+  assert.equal(report.summary["viewer-cannot-minimize"], 2);
+  assert.equal(report.summary.candidate, 0);
+});
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply tests/audit-pr-cleanup-summary.test.mjs:101]: **Accepted** — fixed in 261ba1b: added test `emits clean when already-minimized items have viewerCanMinimize false` which covers `isMinimized: true && viewerCanMinimize: false` → `clean`, `viewer-cannot-minimize: 0`.

File: docs/idd-comment-minimization.md [hunk 3]
```diff
@@ -296,6 +296,95 @@ The helper applies only safe candidates and reports applied, skipped,
...
+**F4 Cleanup Evidence**
+
+| Field              | Value                                  |
+| ------------------ | -------------------------------------- |
+| Status             | applied / failed / incomplete          |
+| Applied            | N                                      |
+| Failed             | N                                      |
+| Skipped            | N                                      |
+| Permission-blocked | N                                      |
+| Notes              | reason for any failed or skipped items |
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply docs/idd-comment-minimization.md:341]: **Accepted** — fixed in 261ba1b: changed all evidence token instances from `permission-blocked:{N}` to `viewer-cannot-minimize:{N}` to match the summary field name used in `computeReportSummary`.

File: .github/instructions/idd-merge.instructions.md [hunk 4]
```diff
@@ -186,32 +185,66 @@ gate. The active claim must still use your current `{claim-id}`.
...
+     succeeded, partially succeeded, or failed. See
+     `docs/idd-comment-minimization.md` for the evidence comment
+     and cleanup-failure comment formats.
+
+     If the apply `status` is `applied`: proceed to step 3.
+
+     If the apply `status` is `failed` or `incomplete`: use the
+     cleanup-failure comment format instead of the evidence comment
+     format. This is explicit evidence, not a merge gate — the merge
+     already succeeded. Proceed to step 3.
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply .github/instructions/idd-merge.instructions.md]: **Accepted** — fixed in 261ba1b: replaced the contradictory "post one evidence comment regardless of outcome" with per-outcome directives — `applied` → evidence comment format; `failed`/`incomplete` → cleanup-failure comment format.

File: idd-template/.github/instructions/idd-merge.instructions.md [hunk 5]
```diff
@@ -186,32 +185,66 @@ gate. The active claim must still use your current `{claim-id}`.
...
+
+     If the apply `status` is `failed` or `incomplete`: use the
+     cleanup-failure comment format instead of the evidence comment
+     format. This is explicit evidence, not a merge gate — the merge
+     already succeeded. Proceed to step 3.
+
+   - **`permission-blocked`**: skipped items exist with
+     `viewerCanMinimize: false` and no apply-eligible candidates were
+     found. Post a cleanup-permission-blocked comment to the PR listing
+     the blocked candidates and the count, then proceed to step 3.
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/.github/instructions/idd-merge.instructions.md]: **Accepted** — fixed in 261ba1b: same instruction rewrite applied to the template mirror.

File: idd-template/.github/instructions/idd-merge.instructions.md [hunk 6]
```diff
@@ -186,32 +185,66 @@ gate. The active claim must still use your current `{claim-id}`.
...
+   already-minimized comments and comments the viewer cannot minimize.
+   Re-validate the active claim before each mutation. After GraphQL
+   cleanup, post an evidence comment summarizing the outcome (status,
+   applied count, skipped count with reasons). If the viewer cannot
+   minimize any detected candidates, post a
+   cleanup-permission-blocked comment instead of exiting silently.
+
+   See `docs/idd-comment-minimization.md` for the evidence comment
+   format, cleanup-failure comment format, permission-blocked comment
+   format, and fallback GraphQL commands.
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/.github/instructions/idd-merge.instructions.md:245]: **Rejected** — the template file is an exact copy of `docs/idd-comment-minimization.md` (as required by `audit/sync-manifest.json` exact-mode). The evidence comment formats are defined inline in that same doc, which the template now carries verbatim.

File: scripts/audit-pr-cleanup-summary.mjs [hunk 7]
```diff
@@ -15,7 +15,13 @@ export function computeReportSummary(report) {
   };
 
   if (report.mode === "dry-run") {
-    report.status = report.candidates.length === 0 ? "clean" : "needs-apply";
+    if (report.candidates.length > 0) {
+      report.status = "needs-apply";
+    } else if (viewerCannotMinimize > 0) {
+      report.status = "permission-blocked";
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply scripts/audit-pr-cleanup-summary.mjs:23]: **Accepted** — fixed in 261ba1b (same root cause as the first thread): `viewerCannotMinimize` now only counts `!isMinimized && !viewerCanMinimize` items.

File: idd-template/.github/instructions/idd-merge.instructions.md [hunk 8]
```diff
@@ -186,32 +185,66 @@ gate. The active claim must still use your current `{claim-id}`.
...
 
-     ```sh
-     node scripts/audit-pr-cleanup.mjs --pr <pr-number> --dry-run --format table
-     ```
+   **Mandatory apply decision tree** — follow this sequence; no path
+   may exit without a recorded reason when cleanup candidates exist:
+
+   In the `{{REPO_NAME}}` source repository, run the helper in dry-run
+   mode first. In adopter repositories, skip to the GraphQL fallback
+   below unless the helper scripts were explicitly installed.
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/.github/instructions/idd-merge.instructions.md]: **Accepted** — already fixed in the `fix(template): sync idd-merge and idd-comment-minimization to live versions` commit: removed `{{REPO_NAME}}` from the template and made both files identical.

File: idd-template/docs/idd-comment-minimization.md [hunk 9]
```diff
@@ -296,6 +296,95 @@ The helper applies only safe candidates and reports applied, skipped,
...
+
+### Cleanup evidence comment
+
+Post this comment to the PR after a successful or partial apply. The
+HTML comment token on the first line acts as a stable machine-readable
+marker so a resuming agent can detect whether the evidence was already
+posted:
+
+```markdown
+<!-- idd-cleanup-evidence: {status} applied:{N} failed:{N} skipped:{N} permission-blocked:{N} -->
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/docs/idd-comment-minimization.md]: **Rejected** — the `idd-template` docs are an exact copy of the live docs (required by sync-manifest.json). Replacing `idd-` with `{{PROJECT_MARKER_PREFIX}}` would break the exact-mode constraint. Decoupling template docs from exact-mode sync is a separate design decision; noting it as a follow-up.

**Comment by kurone-kito** [comment]
advisory-wait: claude-sonnet-4-6 claim-2026-05-13T02:00:26Z-430-a5 request 261ba1beb6625f3a8817f38d5e875b18f63b6cb5 2026-05-13T06:11:40Z

_claude-sonnet-4-6: Copilot advisory-wait request — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
_claude-sonnet-4-6: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
_claude-sonnet-4-6: critique baseline — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
**IDD Live Status** — PR #447

| Field | Value |
|-------|-------|
| Phase | F3 merged |
| Claim | claude-sonnet-4-6 / claim-2026-05-13T02:00:26Z-430-a5 |
| Branch | issue/430-feat-idd-enforce-mandatory-f4-cleanup |
| Last checked | 2026-05-13T06:16:09Z |
| Open blockers | none |
| Next action | F4 cleanup then F5 discover |
| Authoritative by | merge commit 4f153bbb96c5510a9181f83ece333afe6c272a6b HEAD 261ba1b |

**Comment by kurone-kito** [comment]
**F4 Cleanup Evidence**

| Field              | Value                                  |
|--------------------|----------------------------------------|
| Status             | applied                                |
| Applied            | 14                                     |
| Failed             | 0                                      |
| Skipped            | 5                                      |
| Permission-blocked | 0                                      |
| Notes              | 4 stale operational markers (OUTDATED), 10 resolved bot review items (RESOLVED). 5 items skipped: 1 failed-CI context, 4 missing IDD disposition markers |
